RocketControlUnitGUI/src/data.ts

```diff
@@ -0,0 +1,110 @@
+  // data.ts
+  import PocketBase from 'pocketbase';
+  import dotenv from "dotenv";
+
+  
+
+  const ADMIN_EMAIL: string = import.meta.env.VITE_EMAIL;
+  const ADMIN_PASSWORD: string = import.meta.env.VITE_PASSWORD;
+
+ console.log(ADMIN_EMAIL);
+
+  const pb = new PocketBase('http://127.0.0.1:8090');
+  pb.autoCancellation(false);
+
+  let isAuthenticated = false;
+
+
+
+  // Function to authenticate the admin user
+  export async function authenticate() {
+    if (!isAuthenticated) {
+      await pb.admins.authWithPassword(ADMIN_EMAIL, ADMIN_PASSWORD);
+      isAuthenticated = true;
+    }
+  }
+
+  type RecordData = { [key: string]: any };
+  export type AllData = { [collectionName: string]: RecordData[] };
+
+  // Fetch paginated data with existing logic
+  export async function fetchPaginatedData(
+    collectionName: string,
+    sendToChart: (data: RecordData[]) => void,
+    batchSize: number = 10
+  ) {
+    console.log(`Fetching data from collection: ${collectionName}`);
+    let page = 1;
+    let hasMoreData = true;
+
+    try {
+      await authenticate(); // Authenticate once before fetching data
```

Comment:
I believe we only need to authenticate once, not every time we want to read/write to/from pocketbase.